browser/javascript/v2/react-app/src/App.js

```diff
@@ -0,0 +1,187 @@
+
+/**
+ * Start by calling ampli.load()
+ *
+ * 'ampli' is the default instance of Ampli()
+ */
+
+/**
+ * When you pull your tracking plan you can use the defaults and call load() without arguments
+ *
+ * This requires connecting your account via `ampli pull`
+ * which will set you API key in the generated Ampli SDK
+ */
+// ampli.load();
+
+/**
+ * OR Specify an environment
+ */
+// ampli.load({ environment: 'development' })
+
+/** OR Provide a specific Amplitude API key */
+// ampli.load({ client: { apiKey: REACT_APP_AMPLITUDE_API_KEY } })
+
+/**
+ * OR Use an existing Amplitude instance
+ * requires "import amplitude from '@amplitude/analytics-browser';"
+ */
+// amplitude.init(REACT_APP_AMPLITUDE_API_KEY, undefined, { ...DefaultOptions, logLevel: "INFO" });
+// ampli.load({ client: { instance } });
+
+/**
+ * OR Use an existing window.amplitude instance from code snippet (not recommended)
+ */
+// const instance = window.amplitude.init(REACT_APP_AMPLITUDE_API_KEY);
+// ampli.load({ client: { instance } });
+
+/**
+ * OR Specify AmplitudeClient 'options'
+ */
+ampli.load({
+  client: {
+    apiKey: REACT_APP_AMPLITUDE_API_KEY,
+    options: { ...DefaultOptions, logLevel: 3 },
+  },
+});
+
+/**
+ * For testing you can disable ampli
+ */
+// ampli.load({
+//   disabled: process.env.IS_TESTING ? true : false,
+// });
+
+/**
+ * Make as many Ampli instances as you want
+ */
+// const ampli2 = new Ampli();
+// ampli2.load({ client: { apiKey: 'api-key-2' } });
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        <img src={logo} className="App-logo" alt="logo" />
+        <h2>Ampli Browser JavaScript Example with React</h2>
+
+        <button onClick={() => ampli.identify(userId, { requiredNumber: 42 })}>
+          Identify
+        </button>
+
+        <button
+          onClick={() => ampli.setGroup("test group", "browser-js-ampli")}
+        >
+          Group
+        </button>
+
+        <button
+          onClick={() =>
+            ampli.groupIdentify("test group", "browser-js-ampli", {
+              requiredBoolean: true,
+            })
+          }
+        >
+          Group Identify
+        </button>
+
+        <button
+          onClick={() =>
+            ampli.track(
+              new EventWithOptionalProperties({ optionalBoolean: true })
+            )
+          }
+        >
+          Event w/ Optional Properties
+        </button>
+
+        <button
+          onClick={() => {
+            ampli.eventWithAllProperties({
+              requiredNumber: 1.23,
+              requiredArray: ["I'm", "required"],
+              requiredBoolean: false,
+              requiredEnum: "Enum1",
+              requiredInteger: 42,
+              requiredString: "Hi!",
+            });
+          }}
+        >
+          Event w/ All Properties
+        </button>
+
+        <button
+          onClick={() => {
+            ampli.eventNoProperties();
```

Comment:
Can you please add some more examples:
1. Setting EventOptions
2. Adding a plugin (Segment, logging)

Reply:
For the examples, I will do them in a separate PR.